hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -78,6 +78,21 @@ class RobotHW : public InterfaceManager
    * \returns True if initialization was successful
    */
   virtual bool init(ros::NodeHandle& root_nh, ros::NodeHandle &robot_hw_nh) {return true;}
+  
+  /** \brief The stop function is called to stop the RobotHW from a
+   * non-realtime thread. Usually on controller manager shutdown.
```

Comment:
> Usually on controller manager shutdown.

Can the stopping of the robot be done in the destructor ?

Reply:
It could be done. I never test it though...

Reply:
That is how we do it in dynamixel_control_hw ([the code](https://github.com/resibots/dynamixel_control_hw/blob/d9126f101524cca582fd4607199489e27d1d88bb/include/dynamixel_control_hw/hardware_interface.hpp#L134)). I assume that it is also how it is expected to be done. Otherwise, this method would have already been implemented in ROS_control.

However, I would consider this `stop` method as the method to be called if we want to temporarily stop the robot, typically when a soft (ROS-based) emergency button is pressed.